src/log.js

```diff
@@ -1,153 +1,389 @@
 'use strict'
 
-const EventEmitter = require('events').EventEmitter
-const unionWith = require('lodash.unionwith')
-const differenceWith = require('lodash.differencewith')
-const flatten = require('lodash.flatten')
-const take = require('lodash.take')
-const Promise = require('bluebird')
 const Entry = require('./entry')
+const mapSeries = require('./map-series')
 
-const MaxBatchSize = 10  // How many items to keep per local batch
-const MaxHistory   = 256 // How many items to fetch on join
-
+/** 
+ * ipfs-log
+ *
+ * @example
+ * // https://github.com/haadcode/ipfs-log/blob/master/examples/log.js
+ * const IPFS = require('ipfs-daemon')
+ * const Log  = require('ipfs-log')
+ * const ipfs = new IPFS()
+ *
+ * ipfs.on('ready', () => {
+ *   const log1 = Log.create(ipfs, ['one'])
+ *   const log2 = Log.create(ipfs, [{ two: 'hello' }, { ok: true }])
+ *   const out = Log.join(ipfs, log2, log2)
+ *     .collect()
+ *     .map((e) => e.payload)
+ *     .join('\n')
+ *   console.log(out)
+ *   // ['one', '{ two: 'hello' }', '{ ok: true }']
+ * })
+ */
 class Log {
-  constructor(ipfs, id, opts) {
-    this.id = id
-    this._ipfs = ipfs
-    this._items = opts && opts.items ? opts.items : []
-
-    this.options = { maxHistory: MaxHistory }
-    Object.assign(this.options, opts)
-    delete this.options.items
-
-    this._currentBatch = []
-    this._heads = []
-    this.events = new EventEmitter()
+  constructor(ipfs, entries, heads) {
+    this._entries = entries || []
+    this._heads = heads || LogUtils._findHeads(this) || []
   }
 
+  /**
+   * Returns the items in the log
+   * @returns {Array<Entry>}
+   */
   get items() {
-    return this._items.concat(this._currentBatch)
+    return this._entries
   }
 
-  get snapshot() {
-    return {
-      id: this.id,
-      items: this._currentBatch.map((f) => f.hash)
-    }
+  /**
+   * Returns a list of heads as multihashes
+   * @returns {Array<string>}
+   */
+  get heads() {
+    return this._heads
   }
 
-  add(data) {
-    if (this._currentBatch.length >= MaxBatchSize)
-      this._commit()
+  /**
+   * Find a log entry
+   * @param {string} [hash] [The multihash of the entry]
+   * @returns {Entry|undefined}
+   */
+  get(hash) {
+    return this.items.find((e) => e.hash === hash)
+  }
 
-    return Entry.create(this._ipfs, data, this._heads)
-      .then((entry) => {
-        this._heads = [entry.hash]
-        this._currentBatch.push(entry)
-        return entry
+  /**
+   * Returns the log entries as a formatted string
+   * @example
+   * two
+   * └─one
+   *   └─three
+   * @param {boolean} [withHash=false] - If set to 'true', the hash of the entry is included
+   * @returns {string}
+   */
+  toString(withHash) {
```

Comment:
You're absolutely right. Gonna remove the flag as this functionality was for debugging initially. A nicely formatted string is all that is needed :)